src/app/(site)/employee/my_page/edit/_actions/updateEmployeeForMyPageAction.ts

```diff
@@ -0,0 +1,66 @@
+"use server";
+import { updateEmployeeForMyPageUseCase } from "@/usecase/updateEmployeeForMyPage/updateEmployeeForMyPageUseCase";
+import type { SubmissionResult } from "@conform-to/react";
+import { parseWithZod } from "@conform-to/zod";
+import { redirect } from "next/navigation";
+import { updateEmployeeSchema } from "../_schema/updateEmployeeSchema";
+
+export type UpdateEmployeeForMyPageActionResult = {
+	success: boolean;
+	message: string;
+	submission: SubmissionResult<string[]>;
+};
+
+export type UpdateEmployeeForMyPageParams = {
+	userId: string;
+	occupationId: number; // stringからnumberに変更
+	hiringType?: string;
+	workLocationId?: number; // 追加
+	meetingMethod?: string;
+	talkableTopics?: string;
+	selfIntroduction?: string;
+	careerDescription?: string;
+	jobDescription?: string;
+	joiningDescription?: string;
+	otherDescription?: string;
+};
+
+export async function updateEmployeeForMyPageAction(
+	prevState: unknown,
+	formData: FormData,
+) {
+	const submission = parseWithZod(formData, {
+		schema: updateEmployeeSchema,
+	});
+
+	if (submission.status !== "success") {
+		return submission.reply();
+	}
+
+	const employeeData: UpdateEmployeeForMyPageParams = {
+		userId: formData.get("userId") as string,
```

Comment:
セキュリティ上、クライアントコンポーネントからemployeeUserIdをformDataで渡すのではなく、サーバーサイド（今回の場合はserver action）でemployeeUserIdを取得してあげる。

```ts
const userId = getEmployeeUserId();
```

```suggestion
userId: employeeUserId
```

参考：https://zenn.dev/moozaru/articles/d270bbc476758e#server-actions-%E3%81%AE%E5%BC%95%E6%95%B0%E3%81%AB%E6%B3%A8%E6%84%8F